java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -476,9 +481,19 @@ public boolean fileCanBeSkipped(InputFile inputFile) {
         );
         alreadyLoggedSkipStatus = true;
       }
+      contentHashCache.writeToCache(inputFile);
+      return false;
+    }
+    if (!canSkipInContext) {
+      contentHashCache.writeToCache(inputFile);
+      return false;
+    }
+    if (!inputFile.status().equals(InputFile.Status.SAME)) {
+      LOG.trace("File status is: " + inputFile.status() + ". File can be skipped.");
+      contentHashCache.writeToCache(inputFile);
       return false;
     }
-    return canSkipInContext && inputFile.status() == InputFile.Status.SAME;
+    return contentHashCache.hasSameHashCached(inputFile);
```

Comment:
It looks like we are taking a shortcut by checking the status before checking the cache. My understanding is that we should check the cache, and if the cache is disabled, check the status. WDYT?